### PR TITLE
Migrate to Sphinx/RTD hosted docs.

### PR DESCRIPTION
**Pull request by diranged**
Huge WIP. Not ready for merging.

### CODE REVIEW

**Comment by diranged** [comment]
Docs are live here: http://kingpin.readthedocs.org/en/latest/index.html

File: Makefile [hunk 1]
```diff
@@ -37,3 +38,6 @@ kingpin.zip:
 	cd zip; ln -sf kingpin/bin/deploy.py ./__main__.py
 	cd zip; zip -9mrv ../kingpin.zip .
 	rm -rf zip
+
+docs:
+	make -C docs html
```
Comment (siminm) [inline Makefile]: Always use $(MAKE) here and on line 21 -- it'll propagate all the flags and settings along with calling the function

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+.. image:: https://readthedocs.org/projects/kingpin/badge/?version=latest
+.. image:: https://pypip.in/d/kingpin/badge.png
+.. image:: https://badge.fury.io/py/kingpin.png
+
+*Kingpin: the chief element of any system, plan, or the like.*
+
+Kingpin provides 3 main functions:
+
+-  **API Abstraction** - Job instructions are provided to Kingpin via a JSON based DSL (read below). The schema is strict and consistent from one action to another. 
+-  **Automation Engine** - Kingpin is leverages python's `tornado <http://tornado.readthedocs.org>`_ engine.
```
Comment (cmclaughlin) [inline README.rst]: This is in the old readme, but please fix this while you're working on the docs:

    s/Kingpin is leverages/Kingpin leverages/

File: docs/basicuse.rst [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+      -h, --help            show this help message and exit
+      -j JSON, --json=JSON  Path to JSON Deployment File
+      -d, --dry             Executes a dry run only.
+      -l LEVEL, --level=LEVEL
+                            Set logging level (INFO|WARN|DEBUG|ERROR)
+      --debug               Equivalent to --level=DEBUG
+      -c, --color           Colorize the log output
+
+The simplest use cases of this code can be better understood by looking at the
+`simple.json <examples/simple.json>`__ file. Executing it is a simple as this:
```
Comment (cmclaughlin) [inline docs/basicuse.rst]: This link is broken on RTD.

File: docs/basicuse.rst [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+documented in their specific documentation (*see below*).
+
+JSON-based DSL
+~~~~~~~~~~~~~~
+
+The entire model for the configuration is based on the concept of a JSON
+dictionary that contains at least one *actor* configuration. This JSON format
+is highly structured and must rigidly conform to the `JSON Schema <kingpin/schema.py>`__.
+
+*Validation*
```
Comment (cmclaughlin) [inline docs/basicuse.rst]: This doesn't render as a heading on RTD as I think you intended.

Comment (cmclaughlin) [inline docs/basicuse.rst]: Maybe change it to:

Validation: The JSON file will...

File: kingpin/actors/group.py [hunk 5]
```diff
@@ -148,7 +155,81 @@ def _execute(self):
...
+       { "desc": "Clone, then sleep ... then clone, then sleep shorter...",
+         "actor": "group.Sync",
+         "options": {
+           "contexts": [
+             { "ARRAY": "First", "SLEEP": "60", },
+             { "ARRAY": "Second", "SLEEP": "0", }
+           ],
+           "acts": [
+             { "desc":
+               "do something",
```
Comment (siminm) [inline kingpin/actors/group.py]: Unnecessary new line

File: docs/security.rst [hunk 6]
```diff
@@ -0,0 +1,10 @@
+Security
+--------
+
+URLLIB3 Warnings Disabled
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Recently urllib3 library has started issuing
+`InsecurePlatformWarning <https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning>`__.
+We `suppress <kingpin/actors/rightscale/api.py>`__ urllib3 warnings to
```
Comment (siminm) [inline docs/security.rst]: This used to link to our github code

File: docs/basicuse.rst (hunk 3 above)
Comment (diranged) [inline docs/basicuse.rst]: Done!

File: docs/development.rst [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+    import os
+
+    from tornado import gen
+
+    from kingpin.actors import base
+    from kingpin.actors import exceptions
+
+    # All actors must have an __author__ tag. This is used actively
+    # by the Kingpin code, do not forget this!
+    __author__ = 'Billy Joe Armstrong <american_idiot@broadway.com'
```
Comment (siminm) [inline docs/development.rst]: Need the closing `>`

File: docs/basicuse.rst (hunk 4 above)
Comment (diranged) [inline docs/basicuse.rst]: Done!

File: kingpin/actors/group.py (hunk 5 above)
Comment (diranged) [inline kingpin/actors/group.py]: Done

File: docs/development.rst [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+The following options are baked into our *BaseActor* model and must be
+supported by any actor that subclasses it. They are fundamentally critical to
+the behavior of Kingpin, and should not be bypassed or ignored.
+
+``desc``
+''''''''
+
+A string describing the stage or action thats occuring. Meant to be human
+readable and useful for logging. You do not need to do anything intentinally to
+support this option (its handled in *kingpin.actors.base.BaseActor.\ **init**\ ()*).
```
Comment (siminm) [inline docs/development.rst]: Formatting looks weird with the `*`s; should probably be a link to the docs. `it's` should contain an apostrophe

File: docs/security.rst (hunk 6 above)
Comment (diranged) [inline docs/security.rst]: Done

File: docs/development.rst (hunk 8 above)
Comment (diranged) [inline docs/development.rst]: DOne

**Comment by siminm** [comment]
:+1:

**Pull request merged**